packages/foam-core/src/markdown-provider.ts

```diff
@@ -27,15 +30,29 @@ export function createNoteFromMarkdown(
   const id = path.parse(filename).name;
   const tree = parse(markdown);
   let title: string | null = null;
+
   visit(tree, node => {
     if (node.type === 'heading' && node.depth === 1) {
       title = ((node as Parent)!.children[0].value as string) || title;
     }
     return title === null ? CONTINUE : EXIT;
   });
+
   const links: NoteLink[] = [];
   const linkDefinitions: NoteLinkDefinition[] = [];
+  let frontmatter: any = null;
+  let start: Point = { line: 1, column: 1, offset: 0 }; // start position of the note
   visit(tree, node => {
+    if (node.type === 'yaml') {
+      frontmatter = parseYAML(node.value as string);
```

Comment:
I am using the `yaml` package for parsing the frontmatter. Is that okay?

Reply:
The only downside is that it's 27Kb minzipped, and theoretically foam-core could be used in a browser environment, so it adds a bit of chonk to the bundle.

I don't really see a good alternative right now, since I don't want to implement a partial yaml subset parser

Reply:
yeah, I'd say it's probably ok for now, we might want to revisit for optimizations once as part of the browser release